### PR TITLE
MAID-1390 Hole punch server tries to punch a hole for itself using upnp

### PR DESCRIPTION
**Pull request by canndrew**
This shouldn't be merged until we have a response from the rust-igd maintainer about https://github.com/sbstp/rust-igd/pull/15

[Review on Reviewable](https://reviewable.io/reviews/maidsafe/crust/368)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @vinipsmaker

(maidsafe_highfive has picked a reviewer for you, use r? to override)

**Comment by vinipsmaker** [comment]
> This shouldn't be merged until

Tell me when you're ready

**Comment by canndrew** [comment]
This should be good to go (assuming the checks pass)

File: src/hole_punching.rs [hunk 1]
```diff
@@ -240,16 +252,69 @@ impl HolePunchServer {
...
+                                    _                                   => panic!(),
+                                }
+                            }
+                        }
+                    }
+                    Err(e)  => info!("Failed to get external IP using upnp: {:?}", e),
+                }
+                // TODO (canndrew): What is a sensible time to wait between refreshes of our
+                // external ip?
+                ::std::thread::park_timeout_ms(1000 * 60 * 60);
```
Comment (vinipsmaker) [inline src/hole_punching.rs]: Can you put this value as a constant at the beginning of the code?

File: src/map_external_port.rs [hunk 2]
```diff
@@ -111,7 +103,8 @@ fn from_request_result<T>(r: Result<T, igd::RequestError>) -> io::Result<T> {
 fn map_external_port(local_ep: SocketAddrV4, ext_port: ip::Port)
     -> io::Result<ip::Endpoint>
 {
-    let gateway = try!(from_search_result(igd::search_gateway_from(local_ep.ip().clone())));
+    let gateway = try!(from_search_result(igd::search_gateway_from_timeout(local_ep.ip().clone(), 
+                                                                           Duration::from_secs(1))));
```
Comment (vinipsmaker) [inline src/map_external_port.rs]: Put this value as a named constant at the beginning of the code?

**Comment by vinipsmaker** [comment]
Everything else seems fine and ready for merging (the CI failure is related to network problem fetching code).

**Pull request merged**